src/app/wallet/shared/address-table/address-table.component.ts

```diff
@@ -119,16 +121,27 @@ export class AddressTableComponent implements OnInit {
   /** Delete address */
 
   public deleteAddress(label: string, address: string) {
+    this.address = address;
     if (confirm(`Are you sure you want to delete ${label}: ${address}`)) {
       // this._rpc.call(this, 'manageaddressbook', ['del', address], this.rpc_deleteAddress_success);
-      this._rpc.call('manageaddressbook', ['del', address])
-        .subscribe(response => {
-            this.rpc_deleteAddress_success(response);
-          },
-          error => console.log(`${error.error.message}`));
+      if (['Locked', 'Unlocked, staking only'].indexOf(this._rpc.state.get('encryptionstatus')) !== -1) {
+        // unlock wallet and send transaction
+        this._modals.open('unlock', {forceOpen: true, timeout: 3, callback: this.deleteAddressCallBack.bind(this)});
```

Comment:
line is far too long.. Please make it a bit shorter
Also the use of globals here isn't required.
You can create a callback within this scope that already has the address..
this is very unelegant